acnportal/acnsim/models/battery.py

```diff
@@ -188,3 +284,109 @@ def _from_dict(cls, attribute_dict, context_dict, loaded_dict=None):
         )
         cls._from_dict_helper(out_obj, attribute_dict)
         return out_obj, loaded_dict
+
+
+def batt_cap_fn(requested_energy, stay_dur, voltage, period):
+    """ This function takes as input a requested energy, stay duration,
+    and measurement parameters (voltage & period) and calculates the
+    minimum capacity linear 2 stage battery such that it is
+    feasible to deliver requested_energy in stay_dur periods.
+
+    The function returns this minimum total capacity along with an
+    initial capacity, which is the maximum initial capacity the battery
+    with given total capacity may have such that it is feasible (after
+    charging at max rate) to deliver requested_energy in stay_dur
+    periods. Thus, the returned total and initial capacities maximize
+    the amount of time the battery behaves non-ideally during charging.
+
+    For more info on the capacity equations: TODO: point to paper.
```

Comment:
We might want to put a reminder to do this elsewhere since I'm not sure when a document referencing this method will be published. Maybe in the issue tracker.

Reply:
True. I'll make an issue for this. For now, maybe it makes sense to remind ourselves to update this later?

Reply:
Ok, I'll remove the TODO here. The issue serves as enough of a reminder, I think.